cloudamqp/resource_cloudamqp_extra_disk_size.go

```diff
@@ -28,27 +26,103 @@ func resourceExtraDiskSize() *schema.Resource {
 				Required:    true,
 				Description: "Extra disk size in GB",
 			},
+			"allow_downtime": {
+				Type:        schema.TypeBool,
+				ForceNew:    true,
+				Required:    true,
+				Description: "When resizing disk, allow downtime to do so",
+			},
+			"nodes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"disk_size": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Subscription plan disk size",
+						},
+						"additional_disk_size": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Additional added disk size",
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
 func resourceExtraDiskSizeUpdate(d *schema.ResourceData, meta interface{}) error {
-	api := meta.(*api.API)
-	params := make(map[string]interface{})
+	var (
+		api     = meta.(*api.API)
+		params  = make(map[string]interface{})
+		sleep   = 30   // 30 seconds between retires when polling if extra disk size been succesfull.
+		timeout = 1800 // seconds between polling will timeout.
+	)
+
 	params["extra_disk_size"] = d.Get("extra_disk_size")
-	_, err := api.ResizeDisk(d.Get("instance_id").(int), params)
+	params["allow_downtime"] = d.Get("allow_downtime")
+
+	_, err := api.ResizeDisk(d.Get("instance_id").(int), params, sleep, timeout)
 	if err != nil {
 		return err
 	}
 	id := strconv.Itoa(d.Get("instance_id").(int))
 	d.SetId(id)
-	return nil
+
+	return resourceExtraDiskSizeRead(d, meta)
 }
 
 func resourceExtraDiskSizeRead(d *schema.ResourceData, meta interface{}) error {
+	var (
+		api        = meta.(*api.API)
+		instanceID = d.Get("instance_id").(int)
+	)
+
+	data, err := api.ReadNodes(instanceID)
+	if err != nil {
+		return err
+	}
+
+	nodes := make([]map[string]interface{}, len(data))
+	for k, v := range data {
+		nodes[k] = readDiskNode(v)
+	}
+
+	if err = d.Set("nodes", nodes); err != nil {
+		return fmt.Errorf("error setting nodes for resource %s, %s", d.Id(), err)
+	}
+
 	return nil
 }
 
 func resourceExtraDiskSizeDelete(d *schema.ResourceData, meta interface{}) error {
+	// Just remove this resource from the state file, no action taken in backend.
```

Comment:
Maybe make this comment even more clear, WDYT of this?

```suggestion
	// Just remove this resource from the state file, as the delete route does not exist in the backend
	// but we need to allow delete to happen, e.g. when you destroy your instance
```

Reply:
Good, yes is better.